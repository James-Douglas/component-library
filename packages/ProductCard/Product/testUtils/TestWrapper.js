import React from 'react';
import ProductProvider from '../../Context/ProductProvider';

const ProductTestWrapper = ({
  // eslint-disable-next-line react/prop-types
  children, product, onDetails = () => {}, onSelect = () => {},
}) => (
  <ProductProvider product={product} onDetails={onDetails} onSelect={onSelect}>
    { children }
  </ProductProvider>
);

export default ProductTestWrapper;
