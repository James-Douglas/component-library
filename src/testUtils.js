// test-utils.js
import {
  render, fireEvent, wait, act,
} from '@testing-library/react';
import ManorProvider from './providers/ManorProvider';

const customRender = (ui, options) => render(ui, { wrapper: ManorProvider, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export {
  customRender as render,
  // export other used functions to appease eslint
  fireEvent,
  wait,
  act,
};
