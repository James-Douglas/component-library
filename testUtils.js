import React from 'react';
import {
  render, fireEvent, wait, act,
} from '@testing-library/react';
import ManorProvider from './packages/Provider/ManorProvider';

const ManorTestWrapper = ({ children }) => (
  <ManorProvider>
    {children}
  </ManorProvider>
);

const customRender = (ui, options) => render(ui, { wrapper: ManorTestWrapper, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export {
  customRender as render,
  // export other used functions to appease eslint
  fireEvent,
  wait,
  act,
};
