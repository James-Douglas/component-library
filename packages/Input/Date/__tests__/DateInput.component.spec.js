import React from 'react';
import { ctmTheme } from '@comparethemarketau/manor-themes';
import { render, fireEvent } from '../../../../testUtils';
import DateInput from '../DateInput.component';

describe('DateInput', () => {
  it('renders with minimal props', () => {
    const { container } = render(<DateInput id="test-id" handleChange={() => {}} />);
    expect(container.innerHTML).toMatchSnapshot();
  });

  it('renders with lots of props', () => {
    const { container } = render(
      <DateInput
        id="test-id"
        handleChange={() => {}}
        label="test label"
        prefillValue="01/01/2019"
        required={false}
      />,
    );
    const input = container.querySelector('input');
    const id = input.getAttribute('id');
    const label = container.querySelector('label');
    const placeholder = input.getAttribute('placeholder');
    const svg = container.querySelector('svg');
    const inputWrap = container.querySelector('.input-container');

    expect(id).toBe('test-id');
    expect(label.textContent).toBe('test label');
    expect(placeholder).toBe('DD/MM/YYYY');
    expect(svg).toBeInTheDocument();
    expect(input.value).toBe('01/01/2019');

    expect(inputWrap.firstChild).toHaveStyle(`border: ${ctmTheme.borders.prefill}`);
  });

  it('displays input masking', () => {
    const { container } = render(<DateInput id="test-id" handleChange={() => {}} />);

    const input = container.querySelector('input');

    fireEvent.input(input, { target: { value: '1' } });
    expect(input.value).toEqual('1_/__/____');

    fireEvent.input(input, { target: { value: '01' } });
    expect(input.value).toEqual('01/__/____');

    fireEvent.input(input, { target: { value: '011' } });
    expect(input.value).toEqual('01/1_/____');

    fireEvent.input(input, { target: { value: '0101' } });
    expect(input.value).toEqual('01/01/____');

    fireEvent.input(input, { target: { value: '01012' } });
    expect(input.value).toEqual('01/01/2___');

    fireEvent.input(input, { target: { value: '010120' } });
    expect(input.value).toEqual('01/01/20__');

    fireEvent.input(input, { target: { value: '0101200' } });
    expect(input.value).toEqual('01/01/200_');

    fireEvent.input(input, { target: { value: '01012000' } });
    expect(input.value).toEqual('01/01/2000');
  });

  it('calls change handler on change', () => {
    const handler = jest.fn();
    const { container } = render(<DateInput id="test-id" handleChange={handler} />);

    const input = container.querySelector('input');
    fireEvent.input(input, { target: { value: '11/13/2000' } });
    fireEvent.input(input, { target: { value: '11/12/2000' } });

    expect(handler).toHaveBeenCalled();
    expect(handler.mock.calls.length).toBe(2);
    expect(handler.mock.calls[0][0]).toBe('11/13/2000');
    expect(handler.mock.calls[1][0]).toBe('11/12/2000');
  });

  it('calls focus and blur handlers, check suppress class', () => {
    const focusHandler = jest.fn();
    const blurHandler = jest.fn();
    const { container } = render(
      <DateInput
        id="test-id"
        handleFocus={focusHandler}
        handleBlur={blurHandler}
        handleChange={() => {}}
        gtmPidAnonymous
      />,
    );
    const currencyInput = container.querySelector('input');
    fireEvent.focus(currencyInput);
    expect(focusHandler).toHaveBeenCalled();
    fireEvent.blur(currencyInput);
    expect(blurHandler).toHaveBeenCalled();
    const inputField = container.querySelector('.input-container');
    expect(inputField).toHaveClass('data-hj-suppress');
  });

  it('constrains mask based on "format" prop', () => {
    const { container } = render(
      <DateInput
        id="test-id"
        handleChange={() => {}}
        format="MM/YYYY"
        prefillValue="01/2019"
      />,
    );

    const input = container.querySelector('input');
    expect(input.value).toBe('01/2019');

    const placeholder = input.getAttribute('placeholder');
    expect(placeholder).toBe('MM/YYYY');

    fireEvent.input(input, { target: { value: '1' } });
    expect(input.value).toEqual('1_/____');

    fireEvent.input(input, { target: { value: '01' } });
    expect(input.value).toEqual('01/____');

    fireEvent.input(input, { target: { value: '012' } });
    expect(input.value).toEqual('01/2___');

    fireEvent.input(input, { target: { value: '0120' } });
    expect(input.value).toEqual('01/20__');

    fireEvent.input(input, { target: { value: '01200' } });
    expect(input.value).toEqual('01/200_');

    fireEvent.input(input, { target: { value: '012000' } });
    expect(input.value).toEqual('01/2000');

    const inputField = container.querySelector('.input-container');
    expect(inputField).not.toHaveClass('data-hj-suppress');
  });
});
