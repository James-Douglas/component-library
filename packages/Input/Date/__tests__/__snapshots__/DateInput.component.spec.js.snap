// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateInput renders with minimal props 1`] = `"<div class=\\"sc-fzplWN bXErYq input-wrap\\"><div class=\\"sc-AxheI bXnOQb\\"><label id=\\"test-id-label\\" for=\\"test-id\\" class=\\"sc-Axmtr eRRmJW label\\"></label></div><div class=\\"sc-fznZeY gOCnoB input-container\\"><div class=\\"sc-fzokOt bLutgt\\"><span class=\\"sc-fznKkj XyVbL\\" id=\\"test-id-prefix\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"far\\" data-icon=\\"calendar-alt\\" class=\\"svg-inline--fa fa-calendar-alt fa-w-14 sc-fzoLag irDmAM\\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 448 512\\"><path fill=\\"currentColor\\" d=\\"M148 288h-40c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12zm108-12v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm96 0v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm-96 96v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm-96 0v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm192 0v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm96-260v352c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V112c0-26.5 21.5-48 48-48h48V12c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v52h128V12c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v52h48c26.5 0 48 21.5 48 48zm-48 346V160H48v298c0 3.3 2.7 6 6 6h340c3.3 0 6-2.7 6-6z\\"></path></svg></span><div class=\\"sc-fzqBZW jwXeeT input-clear-wrap\\"><input id=\\"test-id\\" name=\\"test-id\\" type=\\"text\\" inputmode=\\"text\\" placeholder=\\"DD/MM/YYYY\\" aria-labelledby=\\"test-id-label\\" aria-describedby=\\"test-id-prefix\\" autocomplete=\\"off\\" maxlength=\\"11\\" class=\\"sc-fzqNJr blqIGP input-default \\" breakpoint=\\"md\\" value=\\"\\"></div></div></div></div>"`;
